src/components/AnalysisModal.js

```diff
@@ -75,10 +86,10 @@ export const AnalysisModal = Utils.withDisplayName('AnalysisModal')(
         }],
         [Utils.DEFAULT, () => Utils.cond(
           [currentTool === tools.RStudio.label || currentTool === tools.Jupyter.label, () => setViewMode(analysisMode)],
-          [currentTool === tools.galaxy.label && !currentApp, () => setViewMode(environmentMode)],
-          [currentTool === tools.galaxy.label && currentApp, () => {
+          [isToolAnApp(currentTool) && !app, () => setViewMode(environmentMode)],
+          [isToolAnApp(currentTool) && !!app, () => {
```

Comment:
@kyuksel FYI since you asked about accessibility. These create tool controls have a couple of serious issues. Right now they are just divs with images inside of them (and the images have no alt text). So the controls are not focusable (meaning that they can't be triggered via the keyboard), and there is no way for someone using a screen reader to know what any of them would actually do. I have created https://broadworkbench.atlassian.net/browse/IA-3043.